src/functions/GaussFunc.cpp

```diff
@@ -279,7 +287,17 @@ double GaussFunc<D>::calcCoulombEnergy(GaussFunc<D> &gf) {
 
 template<int D>
 std::ostream& GaussFunc<D>::print(std::ostream &o) const {
-    o << "Exp:   " << this->getExp() << std::endl;
+    auto expTmp = this->getExp();
+    auto is_array = std::all_of(expTmp.begin(), expTmp.end(),
```

Comment:
Is this concoction checking whether there are elements beyond `expTmp[0]`? It's a bit obfuscated, I think. Since I can't figure out a suggestion for a cleaner way to do it, I'll just say insert a comment here.

The lambda captures `expTmp` by value, but probably capturing `std::begin(expTmp)` is enough.

Reply:
Maybe use [`std::equal`](https://en.cppreference.com/w/cpp/algorithm/equal)?

Reply:
I was not able to make it check if all in the array are the same. Do you have a code example?

Reply:
Actually, after trying it out it doesn't work as I had expected :sweat_smile: